### DISCUSSION TITLE
change bg color of divbased on checkbox state

### DISCUSSION THREAD

**Discussion by jakob-varenbud** [vueform/vueform#290] [Questions] [answered]
hi, i have the following problem: 

 i have created a custom view for a CheckboxgroupCheckbox which is use by my CheckboxGroupElement. In my CheckboxgroupCheckbox I changed the markup and have markup now, which is outside of the vueform classes. So it is simple html and tailwind css. Now my plan is to change some classes in this markup dynamically weather the checkbox is checked or not. 

In plain vue js I would use something like @handle and bind it to my html. This doesnt work here and I did not find a way to manage this. Does somebody know what would be the way here?

**Comment by jakob-varenbud** [comment]
i am working with nuxt.js and tailwind css. usually it would be a simple ternary operator for example

**Comment by jakob-varenbud** [comment]
thought about changing this in my CheckboxGroupElement: 
` :add-classes="
                (form$, el$) => ({
                  CheckboxgroupCheckbox_custom: {
                    container: {
                      'bg-red-500': el$.model,
                      'bg-gray-300': !el$.model,
                    },
                  },
                })
              "`
as I want to see, if there is a change in the el$.model, which is linked to my checkbox.

this doesnt work as well. but the next problem that i would face is, that the divs where the bg-gray-300 originally lays, is not the container, but its just a normal div which i added because of styling purposes in the markup

**Comment by Davvii1** [comment]
HI! can you provide the code fragment of your checkbox and the checkbox group element you trying to change classes dynamically so I can understand more what are you trying to do?

**Reply by jakob-varenbud** [reply]
sure! here is the code fragment of the custom CheckboxgroupCheckbox view where I want to change the bg-gray300 to bg-red-500 conditiontally if the checkbox is checked (it is directly under the slot): 

```
<template>
  <div class="flex h-44 w-full rounded-2xl shadow-sm bg-white">
    <label :class="classes.container">
      <slot
        v-bind="{ name }"
        :classes="classes"
        :is-disabled="isDisabled"
        :id="id"
        :item="item"
        :value="value"
        :items="items"
        :index="index"
      >
        <div
          class="w-full h-4/5 flex flex-col items-center justify-start bg-gray-300 rounded-tl-2xl rounded-tr-2xl"
        >
          <div
            class="text-white rounded-tl-2xl rounded-tr-2xl p-1 justify-center flex w-full"
            v-html="item.label"
          />
          <div class="flex flex-col justify-center items-center">
            <NuxtImg :src="item.img" width="100" height="100" />
          </div>
        </div>
        <div class="h-1/5 flex flex-row pt-2 pl-2">
          <input
            type="checkbox"
            v-model="el$.model"
            v-bind="attrs"
            :value="value"
            :class="classes.input"
            :name="name"
            :id="id"
            :disabled="isDisabled"
          />
          <div :class="classes.description" v-html="item.description" />
        </div>
      </slot>
    </label>
  </div>
</template>

<script>
export default {
  name: "CheckboxgroupCheckbox_custom",
  data() {
    return {
      merge: true,
      defaultClasses: {
        container: "",
        wrapper: "",
        input: "",
        text_wrapper: "",
        text: "",
        description: "",
      },
    };
  },
};
</script>

<style lang="scss"></style>

```
and it is used in the CheckboxgroupElment here, which is also custom as you can see by the name but has the same functionality as the default CheckboxgroupElement: 

```
<CustomCheckboxgroupElement
              name="test"
              field-name="test"
              :rules="['required']"
              :messages="{
                required: 'Text,
              }"
              :override-classes="{
                CheckboxgroupCheckbox: {
                  container: 'w-full h-full',
                },
              }"
              :replace-classes="{
                CheckboxgroupCheckbox: {
                  input_md: {
                    'form-radius-checkbox': 'rounded-full',
                  },
                },
              }"
              :add-classes="{
                CheckboxgroupCheckbox: {
                  description: 'text-sm p-0.5',
                },
              }"
              :items="[
                {
                  value: '0',
                  label: '1',
                  description: '20€',
                  img: './images/test.png',
                },
                {
                  value: '1',
                  label: '2',
                  description: '20€',
                  img: './images/test2.png',
                },
                {
                  value: '2',
                  label: '3',
                  description: '20€',
                  img: './images/test3.png',
                },
                {
                  value: '3',
                  label: '4',
                  description: '20€',
                  img: './images/test4.png',
                },
              ]"
              :views="{
                CheckboxgroupCheckbox: 'cutsom',
              }"
            />
```

**Reply by Davvii1** [reply]
sorry im really trying to understand well, based on what I see, you trying to do this in the container `<label>` tag or in the `<slot>` tag? maybe an approach like you suggested before like a ternary operator or adding that class dynamically to the `<slot>` classes
```vue
<label :class=`${classes.container} ${value ? 'bg-red-500' : 'bg-gray-300'}`>
```
or
```vue
<slot
  v-bind="{ name }"
  :classes="[...classes, value ? 'bg-red-500' : 'bg-gray-300']"
  :is-disabled="isDisabled"
  :id="id"
  :item="item"
  :value="value"
  :items="items"
  :index="index"
>
```
if you want to do the `:add-classes:` approach I suggest maybe you sould add a ref to the element you want to change inside the component as here is 'order_field'
```vue
<CheckboxgroupElement :add-classes="(form$) => ({
  ComponentName: {
    classname: [
      { 'class-value': form$.el$('other_field')?.value === 'some_value' }
    ],
  }
})" ... />
```
Example is here: https://vueform.com/reference/checkboxgroup-element#option-add-classes

or maybe adding the `#element` inside the template tag of the component
```vue
<!-- CustomElement.vue -->

<template>
  <ElementLayout>
    <template #element>
      <input /> <!-- this will render a basic, unstyled input field -->
    </template>
    <!-- ... -->
  </ElementLayout>
</template>
```
Example here: https://vueform.com/docs/creating-elements#template

**Comment by adamberecz** [comment, marked as answer]
I'd recommend checking out how you can use the `$classname` dynamic class when creating elements: https://vueform.com/docs/creating-elements#classes

Eg. if you want to assign dynamic classes based on component state to `wrapper` you can use $wrapper:
```js
defaultClasses: {
  wrapper: 'wrapper-class',
  wrapper_selected: 'is-selected',
  wrapper_unselected: 'is-unselected',
  $wrapper: (classes, ({ el$, value }) => ([
    classes.wrapper,
    el$.value.indexOf(value) !== -1 ? classes.wrapper_selected : classes.wrapper_unselected,
  ]))
}
```

The first arg of the func is the classes object of the component, the second is the component instance where you can access any of its props.

You might also check out other live examples in the source:
https://github.com/vueform/vueform/blob/main/themes/tailwind/classes.js#L1758

Also, we had a very similary discussion on Discord, where the background needed to be dynamic based on the selected cb value. There's a different approach to that similar to what @Davvii1 recommended. Here's an example for that: https://stackblitz.com/edit/github-z3mtvc?file=src%2FApp.vue,vueform.config.js

**Comment by jakob-varenbud** [comment]
hi, thank you for helping out!
this is the way how i solved it: 

```
export default {
  name: "CheckboxgroupCheckbox_custom",
  data() {
    return {
      merge: true,
      defaultClasses: {
        container: "",
        wrapper: "",
        input: "",
        text_wrapper: "",
        text: "",
        description: "",
        dynamicBg:
          "w-full h-4/5 flex flex-col items-center justify-start bg-gray-300 rounded-tl-2xl rounded-tr-2xl",
        dynamicBg_checked:
          "w-full h-4/5 flex flex-col items-center justify-start  rounded-tl-2xl rounded-tr-2xl bg-red-500",
        dynamicBg_unchecked:
          "w-full h-4/5 flex flex-col items-center justify-start bg-gray-300 rounded-tl-2xl rounded-tr-2xl",
        input: "",
        description: "",
        $dynamicBg: (classes, { el$, value }) => [
          classes.dynamicBg,
          el$.value.indexOf(value) !== -1
            ? classes.dynamicBg_checked
            : classes.dynamicBg_unchecked,
        ],
      },
    };
  },
};
```

so i just added `bg-red-500` to the `dynamicBg_checked`.
and binded the div with the`classes.dynamicBg`

```
 <div :class="classes.dynamicBg">
          <div
            class="text-white p-1 justify-center flex w-full"
            v-html="item.label"
          />
          <div class="flex flex-col justify-center items-center">
            <NuxtImg :src="item.img" width="100" height="100" />
          </div>
        </div>
```
